### ISSUE TITLE
Incomplete fields reference detection for Readonly type

### ISSUE THREAD

**Issue by abramobagnara** [Microsoft/TypeScript#19008]
**TypeScript Version:**  2.5.3

**Code**

```ts
interface I {
  f: number;
}

{
  const v: I = { f: 3 };
  const { f } = v;
  console.log(v.f + f);
}

{
  const v: Readonly<I> = { f: 3 };
  const { f } = v; // Here reference to field f is not detected
  console.log(v.f + f);
}
```

**Expected behavior:**

VSCode should show f as a field reference in line marked with comment

**Actual behavior:**

Only other references to f are shown (also for Readonly type).

The problem is particularly severe for its implications in refactoring and references exploring of React code (as components props/state have Readonly type).

I'm not 100% sure if this is a bug that should be posted to Typescript or to VSCode, but I've guessed it belong to a library in Typescript repo. Please inform me if I should close the issue and move it to a different place.

**Comment by andy-ms** [comment]
In `typescript@next`, doing find-all-references on the first declaration of `f` seems to successfully find all uses. Duplicate of #18548?
